src/core/jit/model.rs

```diff
@@ -314,3 +319,58 @@ impl<Input> OperationPlan<Input> {
         self.flat.len()
     }
 }
+
+/// DirectiveAdapter helps to bridge the gap between async_graphql's
+/// ConstDirective and our custom Directive implementation.
+pub trait DirectiveAdapter {
+    fn name(&self) -> &str;
+    fn arguments(&self) -> Vec<(String, &ConstValue)>;
```

Comment:
seems inconsistent to have owned String and reference. Could we return both references and evade string cloning?

Reply:
Also, returning Iterator could be preferred since it's not involving allocation for the vec if not needed